### PR TITLE
Site Domains: Display a card to buy domain with a plan

### LINKED ISSUE #22243: [iOS] Display a card on Site domain
Apply same conditions for displaying plans card on My site dashboard

Image

### PR DESCRIPTION
**Pull request by staskus**
Fixes #22243, #22244, and #22245
Design: rvVgXupiyONUcJMgPSoMFj-fi-3512%3A20137

Display a card to buy domain with a plan in site domains

## Solution

1. Updated `SiteDomainsView` to use `InsetGroup` style which adds insets and rounds corners of cells making SiteDomains view closer to AllDomains view
2. Updated `PresentationCard` for a more narrow SiteDomains case using `DesignSystem`
3. Updated `SiteDomainsView` actions to open different domain selection types depending on a button

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/784bb29f-30d8-4159-814f-e8638adcff26

### Case 1: A site with no domain

1. Open dashboard
2. Select ... (more)
3. Select domains
4. Confirm that a new card with "Upgrade to a plan" and "Just search for a domain" CTAs appear
5. Click "Upgrade to a plan" and confirm that domain selection shows domains for free with a plan
6. Confirm `domains_dashboard_get_plan_tapped` is tracked
7. Click "Just search for a domain" and confirm that domain selection shows domains with full price
8. Confirm `domains_dashboard_get_domain_tapped` is tracked

### Case 2: A site with a plan and no domain

1. Open dashboard
2. Select ... (more)
3. Select domains
4. Confirm that a new card with "Claim your free domain" title and "Search for a domain" CTA
5. Click "Search for a domain" and confirm that domain selection shows domains for free with a plan
6. Click on a domain and confirm that domain registration view appears
7. Confirm `domains_dashboard_add_domain_tapped` is tracked

### Case 3: A site with a domain

1. Open dashboard
2. Select ... (more)
3. Select domains
4. Confirm that new card is not shown, domain card is shown instead
5. Click "Add domain" and confirm that domain selection appears
6. Confirm `domains_dashboard_get_domain_tapped` is tracked

## Regression Notes
1. Potential unintended areas of impact

We can only break existing flows in SiteDomains view so we need to make sure they continue working

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [x] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift [hunk 1]
```diff
@@ -957,6 +959,10 @@ import Foundation
             return "domains_dashboard_viewed"
         case .domainsDashboardAddDomainTapped:
             return "domains_dashboard_add_domain_tapped"
+        case .domainsDashboardGetDomainTapped:
+            return "domains_dashboard_get_domain_tapped"
```
Comment (ravishanker) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:963]: Wasn't this `domains_dashboard_get_domain_tapped` event already there before!

**Review by ravishanker — APPROVED** [review]
Nice work 👍 

Tested it, and is working.  Did only a cursory review of code.  You might need to add release notes before merging.

File: WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift (hunk 1 above)
Comment (staskus) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift:963]: No, it wasn't. I checked Android events and made them match.

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 24.0
 -----
 * [**] [internal] A minor refactor in authentication flow, including but not limited to social sign-in and two factor authentication. [#22086]
+* [***] [Jetpack-only] Plans: Upgrade to a Plan from domains dashboard in Jetpack app [#22261]
```
Comment (guarani) [inline RELEASE-NOTES.txt]: It might be worth adding "WP.com plan" so that no-one (including editorial) thinks this refers to Jetpack plans.

**Review by guarani — APPROVED** [review]
### Case 1: A site with no domain

Works as expected, I see the two new buttons 

### Case 2: A site with a paid plan

Works as expected, I see the buttom to claim a domain

### Case 3: A site with a domain

Here `domains_dashboard_add_domain_tapped` is fired but the testing instructions call for `domains_dashboard_get_domain_tapped`. I think the current behavior is correct so this might be a typo.

**Comment by staskus** [comment]
> Case 3: A site with a domain
Here domains_dashboard_add_domain_tapped is fired but the testing instructions call for domains_dashboard_get_domain_tapped. I think the current behavior is correct so this might be a typo.

Good observation. The behavior is incorrect, the event is correct. `_add_domain_tapped` should be triggered when domain registration is opened. 

For **A site with a domain** we should open domain purchasing, not domain registration.

**Pull request merged**